src/pages/smart-contracts/public-registry-tutorial.md

```diff
@@ -0,0 +1,612 @@
+
+The `todo-registry` contract defines a read-only function `owner-of?` that returns the owner of a registry entry and a data map for details about entries:
+
+```clarity
+(define-read-only (owner-of (registration-id uint)) ... )
+
+(define-map registry
+  ((registry-id uint))
+  (
+    (name (buff 30))
+    (url (buff 255))
+  )
+)
+```
+
+Let's add the owner information and the details for the first ever registered to-do list (with `registry-id` 1) to the `RecentActivities` component. The `SmartContractsApi` of the client library provides methods to read these data from the blockchain.
+
+### Step 1: Add state variable for first registration
+
+At first, add a new state variable `firstRegistration`.
+
+```js
+// src/components/RecentActvities.jsx
+
+const [firstRegistration, setFirstRegistration] = useState();
+```
+
+Then, define a new callback `fetchRegistration` that will contain the calls to the blockchain and update the state variable. Add the following code after the `fetchRegistration` callback and update the effect hook.
+
+```js
+const fetchRegistration = useCallback(async () => {
+  // fetch newest registration
+  // fetch owner
+  // fetch public URL and name
+});
+
+useEffect(() => {
+  fetchActivities();
+  fetchRegistration();
+}, [fetchActivities, fetchRegistration]);
+```
+
+### Step 2: Query owner of the first to-do list
+
+To query the read-only functions of the smart contract, a `SmartContractsApi` object needs to be created, in the same way as the `AccountsApi` object.
+
+```js
+const smartContractsApi = new SmartContractsApi();
+```
+
+Use that api object in the `fetchRegistration` callback to call the read-only method `owner-of?` like this:
+
+```js
+const fetchRegistration = useCallback(async () => {
+  const ownerResponse = await smartContractsApi.callReadOnlyFunction({
+    contractAddress: CONTRACT_ADDRESS,
+    contractName: CONTRACT_NAME,
+    functionName: 'owner-of?',
+    readOnlyFunctionArgs: ReadOnlyFunctionArgsFromJSON({
+      sender: CONTRACT_ADDRESS,
+      arguments: [cvToHex(uintCV(1))],
+    }),
+  });
+  console.log(ownerResponse);
+});
+```
+
+The arguments of the read-only function are provided as an array of hex encoded `ClarityValue`s.
+The helper method `cvToHex` converts a `ClarityValue` into a hex encoded string. It is the reverse function of `hexToCV` that was used to decode the transaction response.
+The `sender` can be any stacks address and is not relevant for the `owner-of?` function.
+
+### Step 3: Deserialize the response
+
+A read-only function call returns a response object with an `okay` property and if the call was okay, it contains a `result` property with the hex encoded Clarity value. Otherwise the `cause` property describes the error. The helper methods `hexToCV` is used again to decode the result. The result is an optional address of type `SomeCV`. It has a property `value` that contains the owner address. Using `cvToString` with `ownerCV.value` returns the address as a string:
```

Comment:
```suggestion
A read-only function call returns a response object with an `okay` property and it's `true`, it contains a `result` property with the hex encoded Clarity value. Otherwise the `cause` property describes the error. The helper methods `hexToCV` is used again to decode the result. The result is an optional address of type `SomeCV`. It has a property `value` that contains the owner address. Using `cvToString` with `ownerCV.value` returns the address as a string:
```